buildpackage/config_reader.go

```diff
@@ -51,6 +52,10 @@ func (r *ConfigReader) Read(path string) (Config, error) {
 		return packageConfig, errors.Errorf("missing %s configuration", style.Symbol("buildpack.uri"))
 	}
 
+	if packageConfig.Platform.OS == "" {
```

Comment:
One pattern I've seen for setting defaults is to set it on the object *before* decoding. Then it will be retained after decoding if nothing overwrote it.